### DISCUSSION TITLE
[Feature Request] Support 'interval' global variable in resample expression

### DISCUSSION THREAD

**Discussion by wolfwfr** [grafana/grafana#48507] [Dashboards, Panels & Visualizations]
Hi,

The resample expression is a very nifty tool, but it would be great if it would be possible to automatically adjust the interval based on grafana's suggested interval in the query options (figure A).

**figure A | grafana's suggested interval**
![grafana interval](https://user-images.githubusercontent.com/37142636/165919311-3f4f4af6-03ee-4f09-9b98-e8dcd3463852.png)

**figure B | current resample interval implementation**
![grafana resample interval - current impl](https://user-images.githubusercontent.com/37142636/165919735-d3afe5b7-0497-4755-a0f5-84b600d6d638.png)

**figure C | suggested resample interval feature**
![grafana resample interval - suggested feature](https://user-images.githubusercontent.com/37142636/165919951-3e375dc7-89b8-4bff-92e5-77ead28ceab4.png)

**Comment by rvdwijngaard** [comment]
+1 for supporting variables for Expressions

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
